implementations/rust/ockam/ockam_api/src/session/mod.rs

```diff
@@ -117,7 +120,11 @@ impl Medic {
                                 let f = session.replacement(session.ping_route().clone());
                                 session.set_status(Status::Down);
                                 log::info!(%key, "replacing session");
-                                self.replacements.spawn(async move { (key, f.await) });
+                                self.replacements.spawn(async move {
+                                    let replacement = (key, f.await);
```

Comment:
The replacer starts from the moment you start waiting for the future, `f.await`.
Here you wait **after** the connection has been attempted, it would be more beneficial to wait **before** the connection attempt, to give time to external services to recover (like a service or a connection restarting).